File: backend/app/schemas/passport_me.py
"""Citizen Passport overview API schemas (PASSPORT-05A)."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class PassportSummaryResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    passport_tier: str | None
    reputation: int
    wallet_balance: int
    earned_badges: int
    active_challenges: int
    claimable_rewards: int


class PassportWalletResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    balance: int
    lifetime_earned: int
    lifetime_spent: int


class PassportReputationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    total_points: int


class PassportOverviewPassportResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    status: str
    created_at: datetime


class PassportBadgeResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    code: str
    name: str
    description: str
    rarity: str
    family: str
    earned_at: datetime | None = None


class PassportChallengeResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    code: str
    name: str
    description: str
    progress: int
    target: int
    completed: bool
    reward_claimed: bool
    ym_reward: int


class PassportOverviewResponse(BaseModel):
    summary: PassportSummaryResponse
    passport: PassportOverviewPassportResponse
    wallet: PassportWalletResponse
    reputation: PassportReputationResponse


class PassportBadgesResponse(BaseModel):
    earned: list[PassportBadgeResponse]
    locked: list[PassportBadgeResponse]


class PassportChallengesResponse(BaseModel):
    active: list[PassportChallengeResponse]
    completed: list[PassportChallengeResponse]
    claimable: list[PassportChallengeResponse]


class ChallengeClaimResponse(BaseModel):
    challenge_code: str
    claimed: bool
    ym_awarded: int
    new_balance: int
    message: str

File: frontend/packages/types/src/passport-me.ts
/** Citizen Passport V2 overview API types (PASSPORT-05A). */

export type PassportSummaryResponse = {
  passport_tier: string | null;
  reputation: number;
  wallet_balance: number;
  earned_badges: number;
  active_challenges: number;
  claimable_rewards: number;
};

export type PassportWalletResponse = {
  balance: number;
  lifetime_earned: number;
  lifetime_spent: number;
};

export type PassportReputationResponse = {
  total_points: number;
};

export type PassportOverviewPassportResponse = {
  status: string;
  created_at: string;
};

export type PassportBadgeResponse = {
  code: string;
  name: string;
  description: string;
  rarity: string;
  family: string;
  earned_at: string | null;
};

export type PassportChallengeResponse = {
  code: string;
  name: string;
  description: string;
  progress: number;
  target: number;
  completed: boolean;
  reward_claimed: boolean;
  ym_reward: number;
};

export type PassportOverviewResponse = {
  summary: PassportSummaryResponse;
  passport: PassportOverviewPassportResponse;
  wallet: PassportWalletResponse;
  reputation: PassportReputationResponse;
};

export type PassportBadgesResponse = {
  earned: PassportBadgeResponse[];
  locked: PassportBadgeResponse[];
};

export type PassportChallengesResponse = {
  active: PassportChallengeResponse[];
  completed: PassportChallengeResponse[];
  claimable: PassportChallengeResponse[];
};

export type ChallengeClaimResponse = {
  challenge_code: string;
  claimed: boolean;
  ym_awarded: number;
  new_balance: number;
  message: string;
};
